docs/zkapps/o1js/bitwise-gadgets.mdx

```diff
@@ -0,0 +1,322 @@
+---
+title: Bitwise Gadgets
+hide_title: true
+description: Bitwise gadgets in o1js.
+keywords:
+  - gadgets
+  - bitwise
+  - o1js
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Bitwise Gadgets in o1js
+
+Bitwise operations work on 2-bit patterns of equal lengths by positionally matching their individual bits.
```

Comment:
Operations are essentially equivalent to functionality.
e.g. I want to know what x AND y is. To find that out, I need to perform an AND operation.

Gadgets are how o1js facilitates that operation.
So in order to perform an AND operation, I use the AND gadget.

So having thought about this some more, the actual functionality is the primary thing we want to communicate in this page so a more appropriate title for the page is Bitwise Operations.
Then we can also mention that the operations are implemented as Gadgets.

Let's discuss this first so that we are aligned on the messaging.